### PR TITLE
Added WikiMedia OneBox thumbnails.

### PR DESCRIPTION
**Pull request by nbianca**
Fixes [this](https://meta.discourse.org/t/image-onebox-breaks-on-wikimedia-links/41197/5).

### CODE REVIEW

**Comment by ZogStriP** [comment]
Awesome 👍 Any chances you could add some tests so that we don't break it in the future?

File: lib/onebox/engine/wikimedia_onebox.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+      def to_html
+        begin
+          match ||= @url.match(/^https?:\/\/commons\.wikimedia\.org\/wiki\/(?<name>File:.+)/)
+          name = match[:name]
+          api_url = "https://en.wikipedia.org/w/api.php?action=query&titles=#{name}&prop=imageinfo&iilimit=50&iiprop=timestamp|user|url&iiurlwidth=500&format=json"
+          result ||= ::JSON.load(open(api_url))
+          pages = result["query"]["pages"]
+
+          ret = {
```
Comment (glebm) [inline lib/onebox/engine/wikimedia_onebox.rb]: Engines must define a `data` method for caching purposes.
A spec would've caught this, please add one.

File: lib/onebox/engine/wikimedia_onebox.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      def self.priority
+        1
+      end
+
+      def to_html
+        begin
+          match ||= @url.match(/^https?:\/\/commons\.wikimedia\.org\/wiki\/(?<name>File:.+)/)
+          name = match[:name]
+          api_url = "https://en.wikipedia.org/w/api.php?action=query&titles=#{name}&prop=imageinfo&iilimit=50&iiprop=timestamp|user|url&iiurlwidth=500&format=json"
+          result ||= ::JSON.load(open(api_url))
```
Comment (glebm) [inline lib/onebox/engine/wikimedia_onebox.rb]: Engines already handle this, all you need to do is define a `url` method, see e.g.:

https://github.com/discourse/onebox/blob/master/lib/onebox/engine/xkcd_onebox.rb

**Comment by nbianca** [comment]
Thank you @ZogStriP and @glebm for your review. I greatly appreciate it.

I hope that everything is ready for merge.

**Comment by ZogStriP** [comment]
Looks good to me 👍

**Comment by erlend-sh** [comment]
Excellent! @nbianca please update the [original topic](https://meta.discourse.org/t/image-onebox-breaks-on-wikimedia-links/41197?u=erlend_sh) to note that your PR has been merged and the issue has been fixed.

**Pull request merged**